### ISSUE TITLE
Unicode completion leaves backslash when variable name overlaps

### ISSUE THREAD

**Issue by michaelaye** [krassowski/jupyterlab-lsp#567]
## Description
When doing \rho<TAB> to get the unicode replacement option for the greek letter rho, AND there are existing variables with "rho" in their name, choosing the unicode letter leaves the backslash in place:

## Reproduce

![image](https://user-images.githubusercontent.com/69774/112418058-66384180-8cee-11eb-986f-ec92692b2961.png)

![image](https://user-images.githubusercontent.com/69774/112418077-72240380-8cee-11eb-9ecf-46282869bd13.png)

## Expected behavior

The greek letter rho to remain without the backslash.

> Note: When there's no name overlap, it works fine!

## Context

- Operating System and version: PopOs (Ubuntu) 20.10
- Browser and version: Brave 1.22.67
- JupyterLab version: 3.0.12 (but have seen this since I use lsp)

<details><summary>Required: installed server extensions</summary>
<pre>
Config dir: /home/x/miniconda3/envs/py38/etc/jupyter
    dask_labextension enabled
    - Validating dask_labextension...
      dask_labextension 5.0.1 OK
    jupyter_lsp enabled
    - Validating jupyter_lsp...
      jupyter_lsp 1.1.4 OK
    jupyter_server_proxy enabled
    - Validating jupyter_server_proxy...
      jupyter_server_proxy  OK
    jupyter_resource_usage enabled
    - Validating jupyter_resource_usage...
      jupyter_resource_usage  OK
    jupyterlab enabled
    - Validating jupyterlab...
      jupyterlab 3.0.12 OK
    jupyterlab_code_formatter enabled
    - Validating jupyterlab_code_formatter...
      jupyterlab_code_formatter 1.4.5 OK
    jupyterlab_git enabled
    - Validating jupyterlab_git...
      jupyterlab_git 0.30.0b2 OK
    nbclassic enabled
    - Validating nbclassic...
      nbclassic  OK
    nbdime enabled
    - Validating nbdime...
      nbdime 3.0.0.b1 OK

Config dir: /usr/local/etc/jupyter
</pre>
</details>

<details><summary>Required: installed lab extensions</summary>
<pre>
JupyterLab v3.0.12
/home/maye/miniconda3/envs/py38/share/jupyter/labextensions
        jupyter-matplotlib v0.9.0 enabled OK
        dask-labextension v5.0.1 enabled OK (python, dask_labextension)
        @krassowski/jupyterlab-lsp v3.5.0 enabled OK (python, jupyterlab_lsp)
        @ryantam626/jupyterlab_code_formatter v1.4.5 enabled OK (python, jupyterlab-code-formatter)
        @pyviz/jupyterlab_pyviz v2.0.1 enabled OK (python, pyviz_comms)
        @jupyterlab/git v0.30.0-beta.2 enabled OK (python, jupyterlab-git)
        @jupyterlab/server-proxy v3.0.2 enabled OK
        @jupyter-server/resource-usage v0.5.0 enabled OK (python, jupyter-resource-usage)
        @jupyter-widgets/jupyterlab-manager v3.0.0 enabled OK (python, jupyterlab_widgets)

Other labextensions (built into JupyterLab)
   app dir: /home/maye/miniconda3/envs/py38/share/jupyter/lab
        @arbennett/base16-solarized-light v0.1.5 enabled OK
        jupyterlab-theme-solarized-dark v2.0.0 enabled OK
        nbdime-jupyterlab v2.1.0-beta.1 enabled OK
</pre>
</details>

**Comment by michaelaye** [comment]
just wondering: how is this a bug and a feature request at the same time: Or does `feature:completion` just indicate in which code segment the bug occurs?

**Comment by krassowski** [comment]
> Or does feature:completion just indicate in which code segment the bug occurs?

Exactly. The code is structured so that each "feature" is a separate class/file (and ultimately will be a separate package); when refactoring a feature I often go over the list of other issues to remember what to consider.

Sorry I wasn't able to get to it so far, but it is on my radar.
